core/coreapi/interface/interface.go

```diff
@@ -38,27 +38,29 @@ type UnixfsAPI interface {
 	Ls(context.Context, Path) ([]*Link, error)
 }
 
-// type ObjectAPI interface {
-// 	New() (cid.Cid, Object)
-// 	Get(string) (Object, error)
-// 	Links(string) ([]*Link, error)
-// 	Data(string) (Reader, error)
-// 	Stat(string) (ObjectStat, error)
-// 	Put(Object) (cid.Cid, error)
-// 	SetData(string, Reader) (cid.Cid, error)
-// 	AppendData(string, Data) (cid.Cid, error)
-// 	AddLink(string, string, string) (cid.Cid, error)
-// 	RmLink(string, string) (cid.Cid, error)
-// }
+//TODO: Should this use paths instead of cids?
+type ObjectAPI interface {
+	New(ctx context.Context) (Node, error)
+	Put(context.Context, Node) error
+	Get(context.Context, Path) (Node, error)
+	Data(context.Context, Path) (io.Reader, error)
+	Links(context.Context, Path) ([]*Link, error)
+	Stat(context.Context, Path) (*ObjectStat, error)
```

Comment:
I've been pondering in #4418 whether we should return a resolved `Path` in almost any place where we already have it resolved internally anyway. In some cases it saves a ton of additional resolve roundtrips, imagine we use the returned `Path` to pass it directly into another Core API function.

This is especially useful in the case of functions that modify existing objects, e.g. SetData or AddLink. If I call it with `/ipfs/QmFoo/a/b/c`, I'd like it to return `/ipfs/QmNewHash/a/b/c`.

(Regardless of that, `Put()` should definitely return a `Path`.)